test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
+    ConfigLoader._instance = None
+    ConfigLoader._instance_ready = False
+
+
+def test_load(monkeypatch):
+    reset_config_loader_singleton()
+    monkeypatch.setenv('VAR3', 'not bar')
+    cfg_loader = ConfigLoader(defaults={'VAR0': 'default value 0',
+                                        'VAR1': 'default value 1'})
+    config = cfg_loader.load('test/foo/lib/config/config.yaml')
+    assert config._config == {'VAR0': 'fizz',
+                              'VAR1': 'default value 1',
+                              'VAR2': 'foo',
+                              'VAR3': 'not bar'}
+    assert config.source == 'test/foo/lib/config/config.yaml'
+
+
+def test_load_missing():
+    reset_config_loader_singleton()
+    cfg_loader = ConfigLoader()
+    with pytest.raises(ConfigLoaderException):
+        cfg_loader.load('test/foo/lib/config/missing-config.yaml')
+
+
+def test_is_a_singleton():
+    reset_config_loader_singleton()
+    assert ConfigLoader(defaults={'VAR0': 'default value 0'}) is ConfigLoader()
```

Comment:
This test **doesn't** check the Singleton implementation. The main goal of singleton pattern is that we have the same 'global' instance for all of the object instances. In those case you can add sth similar to 
```suggestion
    instance1 = ConfigLoader(defaults={'VAR0': 'default value 0'})
    instance2 = ConfigLoader(defaults={'VAR1': 'default value 1'})
    assert instance1 is instance2
```